src/odemis/gui/model/__init__.py

```diff
@@ -596,7 +603,34 @@ def __init__(self, main):
         self.autofocus_active = BooleanVA(False)
 
 
-class LocalizationGUIData(MicroscopyGUIData):
+class CryoGUIData(MicroscopyGUIData):
+    """
+    Represents an interface for handling cryo microscopes.
+    It's meant to be subclassed from the cryo tabs, to share data between all the tabs (like features)
+    """
+    def __init__(self, main):
+        if "cryo" not in main.role:
+            raise ValueError(
+                "Expected a cryo microscope role but found it to be %s." % main.role)
+        MicroscopyGUIData.__init__(self, main)
+        # List VA contains all the CryoFeatures (for the current project).
+        self.features = model.ListVA([])
+
+    def add_new_feature(self, pos_x, pos_y, pos_z=None, f_name=None, milling_angle=None):
+        """
+        Create a new feature and add it to the features list
+        """
+        if not f_name:
+            f_name = new_feature_name(len(self.features.value))
+        if not pos_z:
+            pos_z = self.main.focus.position.value['z']
+        if not milling_angle:
+            milling_angle = DEFAULT_MILLING_ANGLE
+        feature = CryoFeature(f_name, pos_x, pos_y, pos_z, milling_angle)
+        self.features.value.append(feature)
```

Comment:
Let me first ask: is the features bookmarking functionality only accessible in the localization tab? If yes, you could define the features VA list in the localizationGUIData. Similarly for the function "add_new_feature". Because, the chamber class, for example, would have access to this function, although the chamber tab is not relevant to the feature selection and features bookmaking, if i am not mistaken.

Reply:
That's right, the chamber tab was meant to have the overview map view with the features shown. That's why I created this `CryoGUIData` class so any of the tabs would have the features list. My plan was to check on tab creation if any of the other cryo tabs is created -> then point to its features list directly. But it seems that we may not need to add this view to the chamber tab after all and have only the chamber cam view.